src/ontology/reports/sync-subClassOf.added.self_parentage_issues.tsv

```diff
@@ -0,0 +1,83 @@
+subject_mondo_id	subject_mondo_label	object_mondo_id	subject_source_id	object_source_id	object_mondo_label
```

Comment:
## New file: `reports/sync-subClassOf.added.self_parentage_issues.tsv`
This file collects all cases of offending suggested added edges where the same Mondo ID is both the subject and object (see: https://github.com/monarch-initiative/mondo-ingest/issues/400).

Here's a: [google sheet](https://docs.google.com/spreadsheets/d/1kjrzBGxGIirsMbHQWde3u8Z6YZRL0FyUSTM3eDWvEfI/edit#gid=324968868)

### Alternative approaches to this report
a. Keep producing this file as-is, w/ the current implementation.
b. Keep producing this file, but I could change the implementation a bit. Fore example, `sync-subClassOf.direct-in-mondo-only.tsv`is a similar case where a number of sources are included in the same file, but it gets created in a different way.
c. Keep producing, but 1 file for each source instead of / in addition to this single file collating all cases.
d. Keep producing, but add these file(s) to the `.gitignore`
e. No output files; just throw an error whenever this happens.

Optionally, we could go with one of (a) through (d), and also trigger an error to be thrown from the makefile if it detects any cases after it has run subclass sync for all sources.

Reply:
Nice analysis. TBH, it does reveal some problems, no doubt,

I think we should go with (e) but write an info or warn level message into the log in case some debugging is needed.

1. Proxy merges are manually reviewed in Mondo and should therefore be trusted (even if its sometimes questionable - that is another issue). Proxy merges should be reviewed only once (when they are created), and after that, trusted and not reviewed again until a user complains.
1. The script should explicitly exclude subclass statement between classes that are proxy merged (simple filter as you suggest). Fee free to add a logger.warning() just so things like this can be more easily debugged.

I kinda like your table to just be silently generated so it can be reviewed if needed, but it does create churn. Trish should decide if it is needed. Alternatively you could make a one-off review of the 80 or so rows you found and request a curator to confirm the proxy merges once. Again, more churn, so again, Trish decision.

Reply:
Sounds good to me! So I'm going to:
- [ ] 1. Filter these cases from the 'subclass sync - added' output TSV
- [ ] 2. `logger.warning()` when the cases arise
- [ ] 3. Await @twhetzel's determination on your last paragraph in the comment above, but for now will:
  - Move the output from `reports/` to the `.gitignore`'ed `tmp/` dir
  - Opt for no one-off review